schemas/styles/csl-choose.rnc

```diff
@@ -64,6 +64,38 @@ div {
       attribute variable {
         list { variables+ }
       }
+      |
+      condition.atts.complex.dates
+      |
+      condition.atts.complex.general
+
+  
+  # Tests that require two attributes 
+  # Dates
+  condition.atts.complex.dates =
+    ## specifies the variable to be tested
+    attribute tested { 
+      list { variables.dates+ }
+     },
+     (date-precision | date-range)+
+    
+  ## Tests the precision of a date variable
+  date-precision = attribute date-precision { text }
+  
+  ## Tests whether a date variable is within a given range
+  date-range = attribute date-range { text }
+  
+  # General
+  condition.atts.complex.general =
+    ## specifies the variable to be tested
+    attribute tested { 
+      list { variables+ }
```

Comment:
This includes all variables, including dates. Is that what you want?

Reply:
Bumping this. Looks like you should accept this change regardless of whether it fixes the auto-completion issue.

Reply:
Are you sure we should do this? Currenty the idea is to have tests that work for all variables, and some that only work for dates.

Reply:
Two options are to either (1) list the attributes under both general and date or (2) allow general to accept any variable type (including dates). Don't see much of a reason to prefer one over the other, so your call @bdarcus.

Reply:
Yeah currently it's version 2. But the other version is ok as well...

Reply:
So, do you have strong preference for one of the options @bdarcus ?

Reply:
I guess I don't understand the issue. How does this statement square with the implementation?

> Currenty the idea is to have tests that work for all variables, and some that only work for dates.

Reply:
At the moment, we have one complex test that works for all variables: `if tested="X" matches="Y"`.
All the other complex tests are specific for date variables. If I accept your suggestion, I'll have to add `@matches` to the "complex tests for dates" section. Do you prefer that?